### PR TITLE
Feature/kyv 702 last valid purchase date time

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: packages/order-experience-api/src/api/createController.ts [hunk 1]
```diff
@@ -42,7 +43,17 @@ export class CreateController extends AbstractController<typeof requestSchema> {
     req: ValidatedRequest<typeof requestSchema>,
     res: Response
   ): Promise<any> {
-    const { items, customer } = req.body
+    const { items, customer, lastValidPurchaseDateTime } = req.body
+    let currentDateTime = new Date()
+
+    if (
```
Comment (severikupari1) [inline packages/order-experience-api/src/api/createController.ts:49]: Could you create function to this controller which handles this validation logic.

So you can write automation tests easier in later

File: packages/order-experience-api/src/openapi.yaml [hunk 2]
```diff
@@ -499,6 +502,9 @@ paths:
                       type: string
                     namespace:
                       type: string
+                    lastValidPurchaseDateTime:
+                      type: string
+                      format: date-time
```
Comment (severikupari1) [inline packages/order-experience-api/src/openapi.yaml:507]: Could you add the description to here also?

File: packages/payment-experience-api/src/api/createPaymentController.ts [hunk 3]
```diff
@@ -39,6 +44,18 @@ export class CreatePaymentController extends AbstractController<
     } = request
 
     const order = await getOrder({ orderId, user })
+
+    let currentDateTime = new Date()
```
Comment (severikupari1) [inline packages/payment-experience-api/src/api/createPaymentController.ts:48]: Creating function to order-backend and using it here to validate makes this easier, and only needed to change code in one place

File: packages/order-experience-api/src/openapi.yaml (hunk 2 above)
Comment (synthko) [reply packages/order-experience-api/src/openapi.yaml:507]: This is a good idea, also mention of the expected time format.

File: packages/order-experience-api/src/api/confirmAndCreatePayment.ts [hunk 4]
```diff
@@ -39,6 +42,7 @@ export class ConfirmAndCreatePayment extends AbstractController<
     } = req
 
     const order = await confirmOrder({ orderId, user })
+    checkLastValidPurchaseDateTime(order.lastValidPurchaseDateTime)
```
Comment (severikupari1) [inline packages/order-experience-api/src/api/confirmAndCreatePayment.ts:45]: Should this be before we do confirm to order? 

If you need order data, you can fetch it with getOrderAdmin function

**Comment by severikupari1** [comment]
Cant mark this for needs changes to be made! (Because I opened this pr)

**Review by martinldev — APPROVED** [review]
Other than the confirm logic that Severi mentioned, this looks good.

File: packages/order-experience-api/src/api/confirmAndCreatePayment.ts (hunk 4 above)
Comment (synthko) [reply packages/order-experience-api/src/api/confirmAndCreatePayment.ts:45]: As discussed in the daily, I shall change this.

**Pull request merged**